R/classicProcess.R

```diff
@@ -1926,6 +1928,54 @@ ClassicProcess <- function(jaspResults, dataset = NULL, options) {
   }
 }
 
+
+.procRsquared <- function(options, modelsContainer) {
+  # adapted from .semRsquared() from jaspSem/R/sem.R (one table for several models)
+  if (!options[["rSquared"]] || !is.null(modelsContainer[["rsquared"]])) return()
+  
+  # retrieve model objects
+  procResults <- lapply(options[["processModels"]], function(mod) modelsContainer[[mod[["name"]]]][["fittedModel"]]$object)
+  modelNames <- sapply(options[["processModels"]], function(mod) mod[["name"]])
+  
+  # init table
+  tabr2 <- createJaspTable(gettext("R-Squared"))
+  tabr2$addColumnInfo(name = "__var__", title = "", type = "string")
+  for (i in seq_along(options[["processModels"]])) {
+    tabr2$addColumnInfo(name = paste0("rsq_", i), title = modelNames[i],
+                        overtitle = "R\u00B2", type = "number")
+  }
+  
+  tabr2$dependOn(c("rSquared", "processModels"))
+  tabr2$position <- 1
+  
+  modelsContainer[["rsquared"]] <- tabr2
+  
+  if (modelsContainer$getError()) return()
+  
+  # compute data and fill table
+  
+  # retrieve r²
+  r2li <- try(lapply(procResults, lavaan::inspect, what = "r2"))
+  if (inherits(r2li, "try-error")) 
+    return() # return silently
```

Comment:
```suggestion
  r2li <- lapply(procResults, lavaan::inspect, what = "r2")
```

As above, we don't want the table to disappear if any model has an error, but instead exclude those models.